openinfra_core/src/main/java/de/btu/openinfra/backend/db/daos/file/FileDao.java

```diff
@@ -0,0 +1,135 @@
+package de.btu.openinfra.backend.db.daos.file;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+import java.util.UUID;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response.Status;
+
+import de.btu.openinfra.backend.OpenInfraTime;
+import de.btu.openinfra.backend.db.MappingResult;
+import de.btu.openinfra.backend.db.OpenInfraSchemas;
+import de.btu.openinfra.backend.db.daos.OpenInfraDao;
+import de.btu.openinfra.backend.db.jpa.model.file.File;
+import de.btu.openinfra.backend.db.pojos.file.FilePojo;
+import de.btu.openinfra.backend.db.pojos.file.FilesProjectPojo;
+
+public class FileDao extends OpenInfraDao<FilePojo, File> {
+
+	public FileDao() {
+		super(null, OpenInfraSchemas.FILE, File.class);
+	}
+
+	public FileDao(UUID currentProject, OpenInfraSchemas schema) {
+		super(currentProject, schema, File.class);
+	}
+
+	/**
+	 * This method deletes a file from database when the file belongs to the
+	 * user which is logged in.
+	 *
+	 * @param file
+	 * @param subject
+	 * @return
+	 */
+	public boolean delete(UUID file, UUID subject) {
+		FilePojo pojo = read(null, file);
+		if(pojo.getSubject().equals(subject)) {
+			return delete(file);
+		} else {
+			throw new WebApplicationException(Status.FORBIDDEN);

```

Comment:
Better make use of OpenInfraWebException to lead all exceptions through one class.